src/Util/ComposeFileManipulator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+/**
+ * Manipulate Docker Compose Files.
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ * @final
+ */
+class ComposeFileManipulator
+{
+    public const COMPOSE_FILE_VERSION = '3.7';
+
+    private $composeData;
+
+    public function __construct(string $contents = '')
```

Comment:
No need to have a default value I think?

Reply:
Thinking back, we can use the manipulator to create a new file or manipulate an existing one. For the latter we need the contents.. IIRC we went w/ default value to avoid setter injection...

Reply:
We never (except in 2 tests) use `new ComposeFileManipulator()` with no args. If we removed the `= ''` part (so, just `string $contents`), then it just forces us to be explicit - e.g. `new ComposeFileManipulator('')` if you want to create from an empty file.

Reply:
that makes sense.. i was thinking along the lines of avoiding confusion like "this thing needs a string, but i dont have one. hmm i wonder what happens if i pass an empty string.." where as an explicit `= ''` gets our point across better that and empty string is ok..

Reply:
I hear that. From the other side, I think, if an argument is optional, it makes it easier to, sort of, "not notice". I'm cool either way, really

Reply:
open to keeping as is for now, then revisiting this in the next round of make dockers when we have a better idea how this is implemented outside of the single use case we have now?

Reply:
done